feat(rdb): pg16 final touches MTA-5642a (#4476)
scaleway/docs-content: pages/managed-databases-for-postgresql-and-mysql/api-cli/logical-replication-as-subscriber.mdx

    - **Zero-downtime** - Migrate your data without taking the original database offline. Your applications can continue to write data on the original database while the new database is being populated.
    - **Real-time data synchronization** - The subscriber database receives updates in real-time, ensuring that data is consistent across the two databases.
    - **Flexible migration windows** - You can choose when to switch your application to the new database, without having to worry about data consistency. Logical replication ensures that the new database is always up-to-date with the latest changes in the original database.
    - **Easy rollback** - If you encounter issues during migration, such as data corruption and application issues, you can point your application back to the original database. Since the subscriber database constantly keeps a copy of the data in sync with the publisher, you can avoid data loss or inconsistencies.

## How to set up the subscription

To be able to create a subscription, you must have the privileges of the `pg_create_subscription` role, as well as CREATE privileges on the current database.

> 1. Grant `pg_create_subscription` and CREATE privileges to your role of choice. In this example we use `my_replication_user`. Make sure you replace all variables with the information of your databases.

Reviewer: You need an admin user to GRANT pg_create_subscription to a user

    <Message type="tip">
      Refer to the official [Create Publication](https://www.postgresql.org/docs/16/sql-createpublication.html) PostgreSQL documentation to learn how to detail the specific tables or schemas.
    </Message>

    ```sql
    CREATE PUBLICATION my_publication FOR ALL TABLES;
    ```
3. Subscribe the subscriber database to the publication created in the previous step. This will create a subscription that connects both databases.
    <Message type="important">
      Subscriptions in the same database cannot have the same name.
    </Message>

    ```sql
    CREATE SUBSCRIPTION my_subscription
      CONNECTION 'host=publisher_host port=5432 user=my_replication_user password=my_password dbname=my_database'
      PUBLICATION my_publication;
    ```
4. Run the commands below to respectively grant the user read access to the subscription, and to check its statistics. This helps you make sure the subscription was successful. You can check the number of pending changes and last error message in the output.
    ```sql
    GRANT pg_read_all_stats TO my_replication_user;
    SELECT * FROM pg_stat_subscription;
    ```

> 5. Run the first command to grant the user read access to all data. Run the second to get a table containing data about all subscriptions defined in the database. This data includes the subscription ID, name, owner, whether the subscription is enabled or not, and connection information.

Reviewer: ```suggestion
5. Run the first command to grant the user read access to all data. Run the second to get a table containing data about all subscriptions defined in the database. This data includes the subscription ID, name, owner, whether the subscription is enabled or not.
```

    ```sql
    GRANT pg_read_all_data TO myuser;
    SELECT * FROM pg_subscription;
    ```
    <Message type="tip">

>       Refer to the official PostgreSQL documentation for more information on the `pg_subscription` catalog.

Reviewer 2: Maybe link to the official doc

    </Message>

## Limitations

When creating a subscription from a Scaleway Database Instance to a public IP, the public interface of the Database Instance node is used to connect to the publisher. This might incur in the following consequences:

> - If you only have a private endpoint, your Database Instance will still has a public management IP address. The ports are all closed, stopping all connections except the one with the publisher or subscriber database.

Reviewer 3: Only the publisher database -- we don't connect to subscriber databases yet

Also maybe remind that data is securely encrypted

Author: ```suggestion
- If you only have a private endpoint, your Database Instance will still have a public management IP address. The ports are all closed, stopping all connections except the one with the publisher database. All outgoing data is securely encrypted. 
```

> - If you only have a private endpoint, your Database Instance will still has a public management IP address. The ports are all closed, stopping all connections except the one with the publisher or subscriber database.

Reviewer 2: ```suggestion
- If you only have a private endpoint, your Database Instance will still have a public management IP address. The ports are all closed, stopping all connections except the one with the publisher or subscriber database.
```

> - If your Database Instance set-up is composed of more than one node, your IP address is subject to change. In case of failover, for example, the IP address connecting to the publisher database will change. We cannot predict the new IP your nodes might take one. However, you can use [Scaleway's Autonomous System](https://ipinfo.io/AS12876) to check the number (ASN) of your primary node's IP address, look up the IP address allocation for the ASN. This can help you predict the IP address of your failover nodes.

Reviewer 2: ```suggestion
- If your Database Instance set-up is composed of more than one node, your IP address is subject to change. In case of failover, for example, the IP address connecting to the publisher database will change. We cannot predict the new IP your nodes might take. However, you can use [Scaleway's Autonomous System](https://ipinfo.io/AS12876) to check the number (ASN) of your primary node's IP address, and look up the IP address allocation for the ASN. This can help you predict the IP address of your failover nodes.
```
